snooty/parser.py

```diff
@@ -481,6 +498,22 @@ def validate_doc_role(self, node: docutils.nodes.Node) -> None:
             msg = f'"{node["name"]}" could not open "{resolved_target_path}": No such file exists'
             self.diagnostics.append(Diagnostic.error(msg, util.get_line(node)))
 
+    def validate_toctree(
```

Comment:
Makes more sense!

Reply:
Sure, will change the function name!

Would it be odd to pass in `self.diagnostics` when it could just be a method of the class, though? Keeping the function as is would also resemble the other `validate` function!